mobile/src/src/components/Create/CreateCommunity/CommunityContainer.jsx

```diff
@@ -1,19 +1,493 @@
+  PollsEndpoint,
+  PostsEndpoint,
+  ProfileEndpoint,
+} from '../../../utils/endpoints';
+import { Card, IconButton, Button, Portal, Modal } from 'react-native-paper';
+import DateTimePicker from '@react-native-community/datetimepicker';
 
-const CommunityContainer = () => {
-  return (
-    <View styles={styles}>
-      <Text>Create a post in community!</Text>
-    </View>
-  );
-};
+import CreatePageTextInput from '../common/CreatePageTextInput';
+import ButtonWithDownArrow from '../common/ButtonWithDownArrow';
+
+class CommunityContainer extends React.Component {
+  state = {
+    user: { profile_photo: null },
+    selectedPostType: 'Post',
+    setSelectedPostType: (newValue) => {
+      this.setState({ selectedPostType: newValue });
+    },
+    community: null,
+    title: '',
+    body: '',
+    pollOptions: [],
+    eventTime: new Date(),
+    eventLocation: '',
+    eventAdmissionPrice: '',
+    showDatePicker: false,
+    showTimePicker: false,
+  };
+
+  resetPage = () => {
+    this.setState({
+      title: '',
+      body: '',
+      pollOptions: [],
+      eventTime: new Date(),
+      eventLocation: '',
+      eventAdmissionPrice: '',
+      showDatePicker: false,
+    });
+  };
+
+  componentDidMount = async () => {
+    await this.getProfile();
+    this.setState({ community: this.state.user.community });
+  };
+
+  getProfile = async () => {
+    await ProfileEndpoint.get()
+      .then((response) => {
+        this.setState({ user: response.data });
+      })
+      .catch((error) => {
+        console.log('error:', error);
+        console.log('error response:', error.response);
+      });
+  };
+
+  submitPost = async () => {
+    const basicData = {
+      title: this.state.title,
+      content: this.state.body,
+      community: this.state.user.community,
+      tagged_users: [], // TODO add tag user functionality
+    };
+
+    if (this.state.selectedPostType === 'Post') {
+      // Submit post to post endpoint
+      try {
+        let response = await PostsEndpoint.post(basicData);
+        this.resetPage();
+      } catch (error) {
+        // TODO: proper error handling - display modal or smth?
+        console.log('error message:', error.message);
+        console.log('error response:', error.response);
+      }
+    } else if (this.state.selectedPostType === 'Poll') {
+      try {
+        const response = await PollsEndpoint.post(basicData);
+
+        for (const i in this.state.pollOptions) {
+          let optionData = {
+            content: this.state.pollOptions[i],
+            creator: this.state.user.id,
+            poll: response.data.id,
+          };
+
+          // For performance reasons, I am not awaiting this, so there's a chance that the poll
+          // options might arrive in a different order. This doesn't matter at the time of writing this,
+          // but in case it happens and you're wondering why, this might be it.
+          PollOptionsEndpoint.post(optionData);
```

Comment:
On second thought, is it better to `await` this?

Reply:
@shashankanandkannan yep